src/codewhisperer/commands/basicCommands.ts

```diff
@@ -132,9 +132,12 @@ export const showFreeTierLimit = Commands.declare('aws.codeWhisperer.freeTierLim
     vscode.env.openExternal(vscode.Uri.parse(CodeWhispererConstants.learnMoreUri))
 })
 
-export const updateReferenceLog = Commands.declare('aws.codeWhisperer.updateReferenceLog', () => () => {
-    ReferenceLogViewProvider.instance.update()
-})
+export const updateReferenceLog = Commands.declare(
+    { id: 'aws.codeWhisperer.updateReferenceLog', logging: false },
+    () => () => {
+        ReferenceLogViewProvider.instance.update()
+    }
```

Comment:
One thing the logs made obvious, is that this is doing stuff before the extension has had a chance to initialize. This declare() should probably be done in activate() somewhere rather than module-load time?

Reply:
Probably out of scope for this PR though. But wanted to note it since we won't notice it in the logs now.

Reply:
> These declare() calls should probably be done in activate() somewhere, not at module-load?

`declare` doesn't register the command. Registration still happens in `activate`, `declare` simply allows you to reference commands in a type-safe way.

Reply:
Ah ok so those errors we were seeing was because some CW commands are only registered in `activate` but they may be invoked _before_ activation. Module-level registration would stop those errors from happening because it guarantees registration on initialization.

I don't see many downsides to registering commands on module-load because:
* Registering commands is cheap
* There is little need to register different functions dynamically on start-up
* Command registration doesn't trigger side-effects besides binding a function to an id

Commands are an extremely important part of the VS Code API, and to me it feels odd to hide them away in a single function (per-module). Also, if we _really_ need to, we can always defer actual command registration until extension initialization. The `Commands` class can simply wait to call `vscode.registerCommand` until told to do so.